### PR TITLE
ARC: linker: merge GNU and MWDT linker scripts

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
As discussed in #22668 and #28083, there is additional risk associated with splitting linker files, as one may update one script and not be aware of the other. Especially related to updating GNU ld, and not MWDT could break code for MWDT unnoticed, as MWDT is not part of Zephyr CI.

Let's create a single entry point for linker template.

### CODE REVIEW

**Review by abrodkin — APPROVED** [review]

File: include/arch/arc/v2/linker.ld [hunk 1]
```diff
@@ -4,9 +4,231 @@
...
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rodata.ld>
+
+#include <linker/kobject-rom.ld>
+	} GROUP_LINK_IN(ROMABLE_REGION)
+
+#ifdef __MWDT_LINKER_CMD__
+/* TODO: add mwdt specific ROM C++ sections */
```
Comment (ruuddw) [inline include/arch/arc/v2/linker.ld:118]: If for mwdt something special is required, please create an issue so we don't forget to take care of this.
(or, fix it...)

File: include/arch/arc/v2/linker.ld [hunk 2]
```diff
@@ -4,9 +4,231 @@
...
+
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
+
+	__data_rom_start = LOADADDR(_DATA_SECTION_NAME);
+
+#include <linker/common-ram.ld>
+#include <linker/kobject.ld>
+
+#ifdef __MWDT_LINKER_CMD__
+/* TODO: add mwdt specific RAM C++ sections */
```
Comment (ruuddw) [inline include/arch/arc/v2/linker.ld:195]: see above. Also, please add an error message so users get notified C++ isn't supported yet for mwdt.

File: include/arch/arc/v2/linker.ld [hunk 3]
```diff
@@ -4,9 +4,231 @@
...
+	_end = .; /* end of image */
+
+	__kernel_ram_end = .;
+	__kernel_ram_size = __kernel_ram_end - __kernel_ram_start;
+
+#ifdef __MWDT_LINKER_CMD__
+/* mwdt requires _fstack, _estack which will be used in _stkchk.
+ * _stkchk is inserted by mwdt automatically, if _fstack, _estack is not
+ * set correctly, the  brk_s instruction will be called
+ * here, we use a trick to deceive the compiler.
```
Comment (ruuddw) [inline include/arch/arc/v2/linker.ld:213]: When does mwdt call this function? Always, or only if some stack checking is enabled? On every function call?
okay for now, but this seems something that could be enhanced later (new enhancement issue?).

File: include/linker/kobject-text.ld [hunk 4]
```diff
@@ -32,11 +32,11 @@
...
-through Kconfig either disable 'CONFIG_USERSPACE', or set
-'CONFIG_KOBJECT_TEXT_AREA' to a value larger than
+"The configuration system has incorrectly set \
+'CONFIG_KOBJECT_TEXT_AREA' to \
+CONFIG_KOBJECT_TEXT_AREA, which is not big enough. You must \
+through Kconfig either disable 'CONFIG_USERSPACE', or set \
+'CONFIG_KOBJECT_TEXT_AREA' to a value larger than \
 CONFIG_KOBJECT_TEXT_AREA."
 		);
 #endif /* CONFIG_USERSPACE */
```
Comment (ruuddw) [inline include/linker/kobject-text.ld:42]: see sanitycheck: "Please remove blank lines at end of 'include/linker/kobject-text.ld'"

Comment (evgeniy-paltsev) [reply include/linker/kobject-text.ld:42]: Hmm, it's strange why sanitycheck detects this as I don't change that part of file. However I's fixed now.

**Review by ruuddw — APPROVED** [review]
Great, the unified file doesn't have too many #ifdef variation points for mwdt.

**Pull request merged**